lib/src/authentication/src/client_options.dart

```diff
@@ -19,11 +19,6 @@ class ClientOptions extends AuthOptions {
   /// https://docs.ably.com/client-lib-development-guide/features/#TO3a
   String? clientId;
 
-  /// Custom log handler
-  ///
-  /// https://docs.ably.com/client-lib-development-guide/features/#TO3c
-  LogHandler? logHandler;
```

Comment:
I'm wondering if there might be any customers who are currently using this interface, despite the fact that it wasn't actually implemented. Perhaps an alternative approach, given we're only going to be doing the next release of this plugin as a patch version bump, might be:

1. **For the compiler**: Annotate both this instance variable and the `LogHandler` class as deprecated
2. **At runtime**: Fail as fast as possible if a value is assigned to it (either on setting this member if that's possible, or when these client options are supplied to us)

i.e. here, something like:

```dart
@Deprecated('Not used, as log messages are handled by the default mechanism in the underlying SDK. This instance variable will be removed in future.') LogHandler? logHandler;
```

This might just be a gentler approach for those who get automatically upgraded to the version of the SDK that incorporates this change because it's come in via SemVer patch.

Reply:
@QuintinWillison I thought about this, and there's a reason I didn't do this - I think introducing deprecation for a feature that never actually worked may lead to more confusion than simply removing it. If you prefer the deprecation approach in cases like that, I'll restore this and add `@deprecated` warning as you requested 🙂

Reply:
@QuintinWillison It's done in https://github.com/ably/ably-flutter/commit/19de88ca1fdb39cc236fd6e71f7e4fae157ae89c 🙂 I've also added a deprecation warning to the `LogHandler` type itself

Reply:
@QuintinWillison It's done in https://github.com/ably/ably-flutter/commit/19de88ca1fdb39cc236fd6e71f7e4fae157ae89c 🙂 I've also added a deprecation warning to the `LogHandler` type itself

Reply:
Thanks. I have at least one follow-up comment but will submit that in a second review pass.